src/main/webapp/app/exercises/text/assess-new/text-submission-assessment.component.ts

```diff
@@ -30,6 +30,7 @@ import { StructuredGradingCriterionService } from 'app/exercises/shared/structur
 export class TextSubmissionAssessmentComponent implements OnInit {
     private userId: number | null;
     exerciseId: number;
+    courseId: number;
```

Comment:
I think it should not be a getter because the exercise (where we get the course from) is not always initialised. I've set it to 0 in the resetComponent().